cmd/attack/jvm.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attack
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"go.uber.org/fx"
+
+	"github.com/chaos-mesh/chaosd/cmd/server"
+	"github.com/chaos-mesh/chaosd/pkg/core"
+	"github.com/chaos-mesh/chaosd/pkg/server/chaosd"
+	"github.com/chaos-mesh/chaosd/pkg/utils"
+)
+
+func NewJVMAttackCommand() *cobra.Command {
+	options := core.NewJVMCommand()
+	dep := fx.Options(
+		server.Module,
+		fx.Provide(func() *core.JVMCommand {
+			return options
+		}),
+	)
+
+	cmd := &cobra.Command{
+		Use:   "jvm <subcommand>",
+		Short: "JVM attack related commands",
+	}
+
+	cmd.AddCommand(
+		NewJVMAttachCommand(dep, options),
+		NewJVMInstallRuleCommand(dep, options),
+	)
+
+	return cmd
+}
+
+func NewJVMAttachCommand(dep fx.Option, options *core.JVMCommand) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "attach [options]",
+		Short: "attach agent to Java process",
+		Run: func(*cobra.Command, []string) {
+			options.Type = core.JVMAttachType
+			options.CompleteDefaults()
+			utils.FxNewAppWithoutLog(dep, fx.Invoke(jvmCommandFunc)).Run()
+		},
+	}
+
+	cmd.Flags().IntVarP(&options.Port, "port", "", 9288, "the port of agent server")
+	cmd.Flags().IntVarP(&options.Pid, "pid", "", 0, "the pid of Java process which need to attach")
+
+	return cmd
+}
+
+func NewJVMInstallRuleCommand(dep fx.Option, options *core.JVMCommand) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "install [options]",
+		Short: "install rules for byteman agent",
+		Run: func(*cobra.Command, []string) {
+			options.Type = core.JVMInstallRuleType
+			options.CompleteDefaults()
+			utils.FxNewAppWithoutLog(dep, fx.Invoke(jvmCommandFunc)).Run()
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.Name, "name", "n", "", "rule name, should be unique, and will generate one automatically if it is empty")
+	cmd.Flags().StringVarP(&options.Class, "class", "c", "", "Java class name")
+	cmd.Flags().StringVarP(&options.Method, "method", "m", "", "the method name in Java class")
+	cmd.Flags().StringVarP(&options.Action, "action", "a", "", "fault action, values can be latency, exception, return or stress")
```

Comment:
I suggest to splite different action in to different command.
And 'gc' should add into description.

Reply:
Good suggestion, done in fba7031